pkg/ios/xcarchive.go

```diff
@@ -0,0 +1,128 @@
+package ios
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// GetXcodeArchiveLocation retrieves the custom archive location for Xcode archives from user preferences.
+// It first attempts to read the custom location from the system defaults. If the location is not set,
+// it returns the default Xcode archive location within the user's home directory.
+//
+// Parameters:
+// - None
+//
+// Returns:
+// - A string containing the path to the Xcode archive location.
+// - An error if the location cannot be determined or if an error occurs during command execution.
+func GetXcodeArchiveLocation() (string, error) {
+	// Command to read the custom archive location from Xcode preferences
+	cmd := exec.Command("defaults", "read", "com.apple.dt.Xcode", "IDECustomDistributionArchivesLocation")
+
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
+
+	// Execute the command and capture the output
+	err := cmd.Run()
+	if err != nil {
+		// If the command fails, check stderr for additional details
+		if strings.Contains(stderr.String(), "does not exist") {
+			// If the key is not set, return the default location for Xcode archives
+			usr, err := user.Current()
+			if err != nil {
+				return "", fmt.Errorf("unable to get current user: %w", err)
+			}
+			// Return default archive location under user's home directory
+			return filepath.Join(usr.HomeDir, "Library", "Developer", "Xcode", "Archives"), nil
```

Comment:
Should we do an exists check here, just to make sure?

Reply:
It wouldn't hurt